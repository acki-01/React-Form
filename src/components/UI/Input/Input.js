/** @format */

import React from 'react';

import classes from 'components/UI/Input/Input.css';

const input = (props) => {
    const inputClasses = [classes.InputElement];
    let validationInfo = '';
    if (props.shouldValidate && props.invalid && props.touched) {
        inputClasses.push(classes.Invalid);
        validationInfo = props.validationInfo;
    }
    let inpulElement = null;
    switch (props.elementType) {
        case 'input':
            inpulElement = (
                <React.Fragment>
                    <input
                        className={inputClasses.join(' ')}
                        {...props.elementConfig}
                        value={props.value}
                        onChange={props.changed}
                    />
                    <p className={classes.ValidationInfo}>{validationInfo}</p>
                </React.Fragment>
            );
            break;
        default:
            inpulElement = (
                <React.Fragment>
                    <input
                        className={classes.InputElement}
                        {...props.elementConfig}
                        value={props.value}
                        onChange={props.changed}
                    />
                    <p>{props.validationInfo}</p>
                </React.Fragment>
            );
    }
    return (
        <div className={classes.Input}>
            <label className={classes.Label}>{props.label}</label>
            {inpulElement}
        </div>
    );
};

export default input;
