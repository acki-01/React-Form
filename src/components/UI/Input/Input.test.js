/** @format */

import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Input from './Input';

configure({ adapter: new Adapter() });

describe('Input component', () => {
    it('should have proper values', () => {
        const validation = {
            required: true,
            minLength: 2,
        };
        const wrapper = shallow(
            <Input
                elementType="input"
                validationInfo="Required, minimum 2 characters"
                valid={false}
                touched={true}
                shouldValidate={true}
                value="a"
                validation={validation}
            />,
        );
        expect(wrapper.find('p').text()).toBe('Required, minimum 2 characters');
    });
    it('should render correctly', () => {
        const wrapper = shallow(<Input />);
        expect(wrapper).toMatchSnapshot();
    });
});
