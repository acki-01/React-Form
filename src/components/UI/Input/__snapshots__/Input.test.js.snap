// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <input />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          className={undefined}
/>,
        <Unknown>
          <input
                    className={undefined}
                    onChange={undefined}
                    value={undefined}
          />
          <p />
</Unknown>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <input
              className={undefined}
              onChange={undefined}
              value={undefined}
/>,
            <p />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": undefined,
              "onChange": undefined,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": Symbol(react.fragment),
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className={undefined}
/>,
          <Unknown>
            <input
                        className={undefined}
                        onChange={undefined}
                        value={undefined}
            />
            <p />
</Unknown>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <input
                className={undefined}
                onChange={undefined}
                value={undefined}
/>,
              <p />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": undefined,
                "onChange": undefined,
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": Symbol(react.fragment),
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
